assets/js/search_results.js

```diff
@@ -50,24 +50,25 @@ async function postPlanForUser() {
 
   const data = await getPlans();
   // the number of plans equals the array length in the JSON result
-  numberOfPlans = data.length;
-  const sourcePlan = data[planIndex];
+  numberOfPlans = data.travel_plans.length;
+  const sourcePlan = data.travel_plans[planIndex];
+  const destination = data.travel_destination;
 
   await fetch(url, {
     method: "POST",
     headers: {
       "Content-Type": "application/json",
     },
-    body: JSON.stringify(planToView(sourcePlan, planIndex)),
+    body: JSON.stringify(planToView(sourcePlan, planIndex, destination)),
   }).catch((err) => console.error(err));
 
   $(this).attr("disabled", "true");
   $(this).parent().attr("title", "saved!");
 }
 
-function planToView(plan, planIndex) {
+function planToView(plan, planIndex, destination) {
   const url = new URL(document.URL);
-  const location = normalizeLocation(url.searchParams.get("location"));
+  const location = normalizeLocation(destination);
```

Comment:
nice.

Reply:
Do we need `url` anymore ?. I see we get the `date` ?

Reply:
that's correct.